Xamarin.PropertyEditing.Mac/Controls/CombinablePropertyEditor.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections;
+using System.Diagnostics;
+using System.ComponentModel;
+
+using Foundation;
+using AppKit;
+using CoreGraphics;
+
+using Xamarin.PropertyEditing.ViewModels;
+using Xamarin.PropertyEditing.Mac.Resources;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	internal class CombinablePropertyEditor<T>
+		: PropertyEditorControl
+	{
+		public CombinablePropertyEditor ()
+		{
+			base.TranslatesAutoresizingMaskIntoConstraints = false;
+
+			UpdateTheme ();
+		}
+
+		public override NSView FirstKeyView => firstKeyView;
+		public override NSView LastKeyView => lastKeyView;
+
+		internal new CombinablePropertyViewModel<T> ViewModel
+		{
+			get { return (CombinablePropertyViewModel<T>)base.ViewModel; }
+			set { base.ViewModel = value; }
+		}
+
+		List<NSButton> combinableList = new List<NSButton> ();
+		bool dataPopulated;
+		NSView firstKeyView;
+		NSView lastKeyView;
+
+		protected override void HandleErrorsChanged (object sender, DataErrorsChangedEventArgs e)
+		{
+			UpdateErrorsDisplayed (ViewModel.GetErrors (e.PropertyName));
+		}
+
+		protected override void SetEnabled ()
+		{
+			foreach (var item in combinableList) {
+				item.Enabled = ViewModel.Property.CanWrite;
+			}
+		}
+
+		protected override void UpdateErrorsDisplayed (IEnumerable errors)
+		{
+			if (ViewModel.HasErrors) {
+				SetErrors (errors);
+			} else {
+				SetErrors (null);
+				SetEnabled ();
+			}
+		}
+
+		protected override void OnViewModelChanged (PropertyViewModel oldModel)
+		{
+			if (!dataPopulated) {
```

Comment:
So when the view model changes, but it's already been loaded once, how does it load the new view model's list of choices?